src/org/zaproxy/zap/extension/tlsdebug/HttpsCallerLauncher.java

```diff
@@ -0,0 +1,119 @@
+package org.zaproxy.zap.extension.tlsdebug;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FilenameFilter;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Pattern;
+
+import org.parosproxy.paros.Constant;
+
+public class HttpsCallerLauncher {
+
+	private ExtensionTlsDebug extension;
+
+	public HttpsCallerLauncher(ExtensionTlsDebug extension) {
+		this.extension = extension;
+	}
+
+	public void startProcess(URL url, String debugStatus) throws IOException {
+
+		ProcessBuilder pb = new ProcessBuilder(getArgumentsList(url, debugStatus));
+		Process p = pb.start();
+
+		Thread tout = new Thread(new StreamController(p.getInputStream()));
+		Thread terr = new Thread(new StreamController(p.getErrorStream()));
+		tout.start();
+		terr.start();
+		try {
+			tout.join();
+			terr.join();
+		} catch (InterruptedException e) {
+			e.printStackTrace();
+		}
+	}
+
+	private List<String> getArgumentsList(URL url, String debugStatus) throws IOException {
+		List<String> argumentsList = new ArrayList<String>();
+		argumentsList.add(System.getProperty("java.home") + "/bin/java");
+		argumentsList.add("-classpath");
+		argumentsList.add(getClasspath());
+		argumentsList.add("-Djavax.net.debug=" + debugStatus);
+		// argumentsList.add("-agentlib:jdwp=transport=dt_socket,server=y,suspend=y,address=1044");
+		argumentsList.add(HttpsCallerProcess.class.getName());
+		argumentsList.add(url.toString());
+		return argumentsList;
+	}
+
+	private String getClasspath() throws IOException {
+		StringBuffer classpath = new StringBuffer(System.getProperty("java.class.path"));
+		// /lang for message bundles
+		classpath.append(";").append(Constant.getZapInstall()).append("lang");

```

Comment:
Should append a colon instead? (It was failing to find the main class as is.)
Also, it should be appended as char.